Sprint-3/quote-generator/quotes.js

```diff
@@ -491,3 +491,19 @@ const quotes = [
 ];
 
 // call pickFromArray with the quotes array to check you get a random quote
+
+function generateNewQuote() {
+  const newQuote = pickFromArray(quotes);
+  const quoteText = newQuote.quote;
+  const quoteAuthor = newQuote.author;
+
+  document.getElementById("quote").innerText = `"${quoteText}"`;
+  document.getElementById("author").innerText = `- ${quoteAuthor}`;
+}
+function setup() {
+  document.getElementById("new-quote").addEventListener("click", () => {
+    generateNewQuote();
+  });
+}
+
+window.onload = setup;
```

Comment:
When i open this on my computer this doesn't work. I think this is because you do not call the function setup, you only call it check there is a click. When you load a page the quote should display!
But yours only displays when the button is clicked :) 
![image](https://github.com/user-attachments/assets/ed412d31-91a7-464f-96ae-129a499c99ca)

Reply:
hi @eliza-an thank you so much for all your comments, specially this one. I thought this line of code: ` window.onload = setup; ` means that call the function setup. can you please elaborate more on how to do that?

Reply:
You are correct. Window.onload will load the setup function when the window is loaded, but look at what the setup function does. 5e only thing the setup function does, is add an event listener , when the button is clicked you will run generateNewQuote. So it’s not the setup function that should run for window.onload, it’s the generateNewQuote function. Does that make sense? Because when the window loads you want the new quote generated, not the event listener added